### PR TITLE
Add/line function

### PR DESCRIPTION
**Pull request by AHReccese**
#### Reference Issues/PRs
#272

### CODE REVIEW

File: art/functions.py [hunk 1]
```diff
@@ -240,6 +241,53 @@ def aprint(artname, number=1, space=1):
...
 
+def lprint(length=15, height=1, char='#'):
+    """
+    Print a grid (length X height) of the given character.
+
+    :param length: the grid length
+    :type length: int
+    :param height: the grid height
+    :type height: int
+    :param char: target character to print a grid of
```
Comment (sepandhaghighi) [inline art/functions.py:252]: target character<s> to print a grid of</s>

File: art/functions.py [hunk 2]
```diff
@@ -240,6 +241,53 @@ def aprint(artname, number=1, space=1):
...
+
+def line(length=15, height=1, char='#'):
+    """
+    Generate a grid (length X height) of the given character.
+
+    :param length: the grid length
+    :type length: int
+    :param height: the grid height
+    :type height: int
+    :param char: target character to generate a grid of
```
Comment (sepandhaghighi) [inline art/functions.py:271]: target character<s> to generate a grid of</s>

File: art/functions.py [hunk 3]
```diff
@@ -240,6 +241,53 @@ def aprint(artname, number=1, space=1):
...
+    :type char: str
+    :return: generated grid as str
+    """
+    if not isinstance(length, int):
+        raise artError(LENGTH_TYPE_ERROR)
+    if not isinstance(height, int):
+        raise artError(HEIGHT_TYPE_ERROR)
+    if not isinstance(char, str) or len(char) != 1:
```
Comment (sepandhaghighi) [inline art/functions.py:279]: Remove `len(char) != 1` (to support non-ascii characters)

File: art/params.py [hunk 4]
```diff
@@ -32,6 +32,11 @@
 FONT_ENVIRONMENT_WARNING = "[Warning] '{0}' font is not printable in this environment."
 FONT_OR_DECOR_ENVIRONMENT_WARNING = "[Warning] '{0}' font or '{1}' decoration is not printable in this environment."
 PACKAGE_LOAD_WARNING = "[Warning] There is a problem loading the package 'coverage'."
+HEIGHT_TYPE_ERROR = "The 'height' type must be int."
+HEIGHT_RANGE_ERROR = "The `height` parameter should be >= 1."
+LENGTH_TYPE_ERROR = "The 'length' type must be int."
+LENGTH_RANGE_ERROR = "The `length` parameter should be >= 1."
+CHAR_TYPE_ERROR = "The 'char' type must be str and has length of 1."
```
Comment (sepandhaghighi) [inline art/params.py:39]: "The 'char' type must be str<s> and has length of 1</s>."

File: art/params.py [hunk 5]
```diff
@@ -32,6 +32,11 @@
 FONT_ENVIRONMENT_WARNING = "[Warning] '{0}' font is not printable in this environment."
 FONT_OR_DECOR_ENVIRONMENT_WARNING = "[Warning] '{0}' font or '{1}' decoration is not printable in this environment."
 PACKAGE_LOAD_WARNING = "[Warning] There is a problem loading the package 'coverage'."
+HEIGHT_TYPE_ERROR = "The 'height' type must be int."
+HEIGHT_RANGE_ERROR = "The `height` parameter should be >= 1."
```
Comment (sepandhaghighi) [inline art/params.py:36]: `height` --> 'height'

File: art/params.py [hunk 6]
```diff
@@ -32,6 +32,11 @@
...
+LENGTH_TYPE_ERROR = "The 'length' type must be int."
+LENGTH_RANGE_ERROR = "The `length` parameter should be >= 1."
```
Comment (sepandhaghighi) [inline art/params.py:38]: `length` --> 'length'

File: art/functions.py [hunk 7]
```diff
@@ -240,6 +241,53 @@ def aprint(artname, number=1, space=1):
...
+        raise artError(CHAR_TYPE_ERROR)
+
+    if length < 1:
+        raise artError(LENGTH_RANGE_ERROR)
+    if height < 1:
+        raise artError(HEIGHT_RANGE_ERROR)
+
+    line_str = char * length
+    return "\n".join([line_str for _ in range(height)])
```
Comment (sepandhaghighi) [inline art/functions.py:288]:
```python
"\n".join([line_str] * height)
```

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
Thanks for your efforts 💯 
1. Here(https://github.com/sepandhaghighi/art/pull/283/commits/90679af1a5f1187035878d56978c372c2b783759), I made some changes to fix tests, so please update your local branch

2. Take a look at my comments

3. Add `lprint` and `line` functions to `__init__.py`

4. Add some tests for `lprint` and `line` at the top of `tests/test.py`

⚠️ We will update `README.md` in another pull request.

**Review by sepandhaghighi — APPROVED** [review]
LGTM 🔥

**Review by sadrasabouri — APPROVED** [review]
LGTM :rocket:

**Pull request merged**